apis/cache/v1alpha1/rediscluster_types.go

```diff
@@ -33,8 +31,6 @@ type RedisClusterSpec struct {
 	EngineVersion string `json:"engineVersion,omitempty"`
 }
 
-var _ resource.Claim = &RedisCluster{}
```

Comment:
I definitely think they're useful for hand written and maintained code, but I'm less sure whether they're useful when all the methods used to satisfy the interface are generated.

I think that if we did want to keep these tests it would make more sense for the code generation tool to add them as part of the generated files, to assert that its intention is to generate methods that satisfy a particular interface.

Reply:
+1 for adding the type checks via generation. I think they are useful for someone trying to understand how these structs are used with `crossplane-runtime`.

Reply:
Looks like @hasheddan raised https://github.com/crossplaneio/crossplane-tools/issues/2 for this.